### ISSUE TITLE
Copy only one sample from the tensor

### ISSUE THREAD

**Issue by don-prog** [davisking/dlib#565]
I'm trying to use dlib framework to define my own loss layer. 
In this loss layer I get a few samples in the output_tensor. Next my loss function should e.g. add vector of the first sample to the vector of the third sample, but I can't do this, because I can't get certain sample separately from other using tensor_tools - tensor_tools::copy_tensor() copies **each sample** to the dest tensor. 

**So how can I extract only one sample from the tensor using dlib's tools?**

**Comment by davisking** [comment]
The tensor has a host() function that gives you a pointer to the data.  You get that then you do whatever you want with the data in the tensor.

**Comment by davisking** [comment]
I should also point out that the tensor interface is completely documented.  http://dlib.net/dlib/dnn/tensor_abstract.h.html#tensor.

**Comment by don-prog** [comment]
@davisking many thanks for your quick response and explanation. 
I already know about host() method using which I can iterate over tensor's data, but I guess dlib's method, which uses CUDA for this task, will be faster. 
What do you think about it, and does this method exist? 
Regards!

**Comment by davisking** [comment]
If you want to write a custom loss then you have to write code to do it.  I don't know what you are trying to do so it should be no surprise it's not in dlib.  If you think you want to run the loss computation on the GPU then write some CUDA code.

**Comment by don-prog** [comment]
@davisking you are saying absolutely right things, I think it's all clear for me. But I'm trying to understand something much more simpler. 
I want to use tt::add() function, for this I need two tensors with num_samples() == 1, so can I extract only one sample from the output_tensor()? I already saw set_sample() function, so maybe I can find something like get_sample()? If I can't do it using tensor, can I convert tensor to matrix using mat() function and then extract only one sample? Or iterating over host() will be only one solution for this purpose at the moment(until I make my own)?
Regards!

**Comment by davisking** [comment]
Maybe you can do what you want with the alias_tensor.

**Comment by don-prog** [comment]
@davisking indeed! alias_tensor works, thanks! I don't want create one more thread in your repo for this small question, so I ask here, if you don't mind:
If I will change class of loss to my custom class and number of output neurons to my custom value in the training and testing definitions of the network, is this enough for using my custom loss with custom number of output neurons? Example:
using net_type = **my_loss**<fc_no_bias<**6**,avg_pool_everything<...>>>

Or I need to change some other settings? 
Regards!

**Comment by davisking** [comment]
That's what you do.

**Comment by don-prog** [comment]
@davisking do you mean this is enough and the settings are correct?

**Comment by davisking** [comment]
Yes

**Comment by don-prog** [comment]
@davisking ok, thanks!
